internal/docker/builder.go

```diff
@@ -211,18 +221,18 @@ func (d *Builder) ConstructBlueprints(bs []b.Blueprint) error {
 }
 
 // construct all Homeservers sequentially then commits them
-func (d *Builder) construct(bprint b.Blueprint, bpWg *sync.WaitGroup) {
-	defer bpWg.Done()
-	networkID := createNetwork(d.Docker, bprint.Name)
-	if networkID == "" {
-		return
+func (d *Builder) construct(bprint b.Blueprint) (errs []error) {
+	networkID, err := createNetwork(d.Docker, bprint.Name)
+	if err != nil {
+		return []error{err}
 	}
 
 	runner := instruction.NewRunner(bprint.Name, d.debugLogging)
 	results := make([]result, len(bprint.Homeservers))
 	for i, hs := range bprint.Homeservers {
 		res := d.constructHomeserver(bprint.Name, runner, hs, networkID)
-		if res.err != nil && res.containerID != "" {
+		if res.err != nil {
```

Comment:
We still presumably want to check `if res.containerID != ""` before calling `printLogs` else if `constructHomeserver` failed before the container was made, there will be no logs to print.

Reply:
Done!